### ISSUE TITLE
Can't cast YouTube videos with FCast while one device is using a VPN

### ISSUE THREAD

**Issue by Blackilykat** [futo-org/grayjay-android#1656]
### What happened?

Whenever I try to cast a YouTube video, it just shows `Source error → Response Code: 403` at the bottom of the screen. However if I stop casting the video plays just fine on my phone. 
The problem does not happen on Nebula videos (the only other source I tried) 

I'm using a VPN on my phone but not on my TV.

### Grayjay Version

278

### What plugins are you seeing the problem on?

Youtube

### Plugin Version

219

### When do you experience the issue?

- [ ] While logged in
- [X] While logged out
- [ ] N/A

### Are you using a VPN?

Yes

**Comment by codebymatter** [comment]
Yeah I had this problem happen to me going as far as Feb 2024. I have a tablet and Fire Stick with Fcast installed.

While your device with Grayjay installed is on a VPN, with your TV not. My phone is not on VPN but my tablet is. Fire Stick is not on VPN.

I can connect to Fire Stick just fine. On tablet, Grayjay would take a while to connect and would encounter an error. It is then that I decided to see if  excluding Fcast frim VPN would work as the infirmation on Fcast showed mismatching IP with my actual IP. (Thats because of the VPN; it assigned an IP address based on region selected from VPN). Abd alas I am able to watch on my tablet.

tldr: both devices have to be on the same network and region in order to pair successfully. You'd have to setup a LAN proxy and update your TVs wi-fi settings to include proxy address. Or if you have a streaming atick that aupports APKs, install yadayada VPN and go from there.

**Comment by Blackilykat** [comment]
That's confusing. I would assume the connection to YouTube is made only on the phone and that the devices are only communicating through the local network (which would be excluded from my VPN anyways since they're private IPs). If my assumption is right I don't see how a VPN would affect anything.
I'll look deeper into it.

**Comment by Blackilykat** [comment]
Well it didn't take much looking to see my assumptions were wrong. It seems like the phone is actually only communicating the url where the tv would load the video. And clearly youtube expects the requests to come from the same IP address.
![image](https://github.com/user-attachments/assets/bf7c358f-e5d8-4abf-abb1-ef74fa23e0e8)

**Comment by VoxelPrismatic** [comment]
There is an option to proxy all requests via your phone
![ResizedImage_2024-12-27_20-32-25_1226.png](https://github.com/user-attachments/assets/a4f247a1-f003-40d5-a0b9-8df7238ca1dc)

**Comment by Blackilykat** [comment]
> There is an option to proxy all requests via your phone 

Thanks! That fixes the problem. I'll leave this open as I think work could still be done to make this more intuitive, but FUTO can close it if they think it's appropriate.

**Comment by Zvonimir-FUTO** [comment]
Opened #1699 to improve this. Closing this bug report.
